clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -303,5 +390,513 @@ def FGenTypeN              : GenericType<"FGenTypeN", TLFloat, VecAndScalar>;
 
 // 2.1. Math extended instructions
 
-def : OCLSPVBuiltin<"acos", [FGenTypeN, FGenTypeN], Attr.Const>;
 
+foreach name  = ["acos", "acosh", "acospi",
+                 "asin", "asinh", "asinpi",
+                 "atan", "atanh", "atanpi",
+                 "cbrt", "ceil", "cos",
+                 "cosh", "cospi",
+                 "erfc", "erf",
+                 "exp", "exp2", "exp10",
+                 "expm1", "fabs", "floor", "lgamma",
+                 "log", "log2", "log10", "log1p", "logb",
+                 "rint", "round", "rsqrt",
+                 "sin", "sinh", "sinpi",
+                 "sqrt",
+                 "tan", "tanh", "tanpi",
+                 "tgamma", "trunc"] in {
+  def : OCLSPVBuiltin<name, [FGenTypeN, FGenTypeN], Attr.Const>;
+}
+
+foreach name  = ["fmax", "fmin", "fmod",
+                 "atan2", "atan2pi",
+                 "copysign", "fdim", "hypot",
+                 "maxmag", "minmag", "nextafter",
+                 "pow", "powr", "remainder"] in {
+  def : OCLSPVBuiltin<name, [FGenTypeN, FGenTypeN, FGenTypeN], Attr.Const>;
+}
+
+foreach name = ["fma", "mad"] in {
+  def : OCLSPVBuiltin<name, [FGenTypeN, FGenTypeN, FGenTypeN, FGenTypeN], Attr.Const>;
```

Comment:
It looks like most of the OCLSPVBuiltin-s has `Attr.Const` attribute.
Does it make sense to add it default and have a separate definition for non-const built-ins if needed?
This should reduce the code.

Reply:
Make sense to do the opposite I think `ConstOCLSPVBuiltin` and leave `OCLSPVBuiltin` as it is.